.github/workflows/push_docker.yaml

```diff
@@ -125,11 +124,16 @@ jobs:
         run: |
           tags=$(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON")
           digests=$(find /tmp/digests -type f -name '*.digest' | xargs cat)
-          cmd="docker buildx imagetools create $tags"
+          cmd_primary="docker buildx imagetools create $tags"
+          for digest in $digests; do
+            cmd_primary="$cmd_primary $REGISTRY_IMAGE@sha256:${digest#sha256:}"
+          done
+          cmd_secondary="docker buildx imagetools create $tags"
           for digest in $digests; do
-            cmd="$cmd $REGISTRY_IMAGE@sha256:${digest#sha256:}"
+            cmd_secondary="$cmd_secondary $ADDITIONAL_REGISTRY_IMAGE@sha256:${digest#sha256:}"
           done
-          eval $cmd 
+          eval $cmd_primary
+          eval $cmd_secondary
```

Comment:
Something is strange. From eyeballing, `tags` should already be `-t mlenergy/zeus:multi-arch-docker-build -t symbioticlab/zeus:multi-arch-docker-build` (containing both repositories) and `digests` would have the images for amd64 and arm64. After that, you're building up two commands (`cmd_primary` and `cmd_secondary`). Did you actually check what the values of `cmd_primary` and `cmd_secondary` are, and confirmed that you're not duplicating work?

<img width="1846" alt="Screenshot 2024-08-04 at 3 30 28 PM" src="https://github.com/user-attachments/assets/ff2464d7-a7ed-485c-b4ec-c1e71d92e4d9">

Actions logs also seem to indicate that each image is getting pushed twice.

Reply:
Sorry for that! I initially thought that DOCKER_METADATA_OUTPUT_JSON was generated from the outputs of the build-push-action and that we needed to create a separate manifest list for each registry. However, I now realize that DOCKER_METADATA_OUTPUT_JSON comes from the metadata-action@v5. Since both registries are included in the images parameter of the metadata-action and symbioticlab/zeus is a mirror registry, the modified setup should work as expected without duplicating the image pushes.